kratos/controllers/controller.h

```diff
@@ -0,0 +1,175 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Daniel Diez
+//
+
+#pragma once
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "containers/model.h"
+#include "includes/kratos_flags.h"
+#include "includes/kratos_parameters.h"
+#include "includes/registry.h"
+
+namespace Kratos
+{
+
+///@name Kratos Classes
+///@{
+
+/**
+ * @class Controller
+ * @ingroup KratosCore
+ * @brief The base  class for all Controllers in Kratos.
+ * @details Controller is the base class for all controllers and defines the interface for them.
+ * Each class must define their own Evaluate method.
+ * Controller parameters must be passed at construction time.
+  @author Daniel Diez
+*/
+class Controller
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of Controller
+    KRATOS_CLASS_POINTER_DEFINITION(Controller);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    explicit Controller() = default;
```

Comment:
```suggestion
    explicit Controller() noexcept = default;
```

also, why `explicit`?

Reply:
no reason, it was like that in operation.h but I think its confusing here.